### PR TITLE
refactor: Remove unused reload config

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Cloud team no need `reload_config` now. cc @flaneur2020 

Fixes #issue

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]
Great!

File: query/src/sessions/session_mgr.rs [hunk 1]
```diff
@@ -409,50 +409,6 @@ impl SessionManager {
         Ok(op)
     }
 
-    pub async fn reload_config(&self) -> Result<()> {
-        // TODO(xp): Potential race condition if fields are updated one by one.
-        //           These fields should all be got prepared then update in one atomic update.
-
-        let config = {
-            let mut config = self.conf.write();
```
Comment (Xuanwo) [inline query/src/sessions/session_mgr.rs:417]: Maybe we can remove the lock of config if we won't reload it?

Comment (ariesdevil) [reply query/src/sessions/session_mgr.rs:417]: Yup

**Review by Xuanwo — APPROVED** [review]
Now we have a lock-free config, cool!

**Comment by BohuTANG** [comment]
This is used by the cloud platform, I added @flaneur2020  and @ZhiHanZ  to review.

**Review by flaneur2020 — APPROVED** [review]

**Comment by flaneur2020** [comment]
LGTM, @ZhiHanZ PTAL

**Review by sundy-li — APPROVED** [review]

**Pull request merged**